wordpress-mobile/WordPress-Android
# Use translatable strings for user roles

Fixes #4496. As explained in the issue, the string arrays outside of `strings.xml` are not actually translated. This PR moves the user roles to `strings.xml` and uses the established pattern for creating the string arrays around it.

In the original implementation, we used these role strings as REST parameters, which is something we should never do 😢 The good thing is since these strings are not translated, it never affected anyone. If they were translated, all role change and invite requests would fail if the language is not English. This PR fixes that and converts string resources to REST parameters with a helper method.

**To test:**
The only way to reproduce the issue is to jump to a commit like 223389f12ce06a1b7964e8adbc8c1decddafda9d and add a role string to your preferred language and try and change the role of a user. As an example, add `role_contributor` to `values-tr/strings.xml` and set it's value anything other than `Contributor`. Set your language to Turkish and change the role of a user to `Contributor`. Now that the translation is not `contributor` anymore, it'll send a different REST param which will return an error. Once you reproduce it, you can test the PR and see that it's not an issue anymore.

@nbradbury I am pinging you as a follow up to #4492, let me know if you don't have time and I'll just set to anyone can review it. Thanks!
